src/controllers/message.controller.js

```diff
@@ -0,0 +1,66 @@
+const { ApiError } = require('../utils/api.error');
+const messagesService = require('../services/message.service');
+const { EventEmitter } = require('ws');
+const messageEmitter = new EventEmitter();
+
+const getMessage = async (req, res) => {
+  res.setHeader('Content-Type', 'text/event-stream');
+  res.setHeader('Cache-Control', 'no-store');
+  res.setHeader('Connection', 'keep-alive');
+
+  const callback = (data) => {
+    res.write(`data: ${JSON.stringify(data)}\n\n`);
+  };
+
+  messageEmitter.on('message', callback);
+  req.on('close', () => messageEmitter.off('message', callback));
+};
+
+const getAllMessages = async (req, res) => {
+  const { roomId } = req.params;
+
+  if (!roomId) {
+    throw ApiError.badRequest('The id for the room is not provided!');
+  }
+
+  const messages = await messagesService.getAllByRoom(roomId);
+
+  if (!messages) {
```

Comment:
The check for `!messages` might not be sufficient. Consider checking if `messages.length === 0` to handle cases where the room exists but has no messages.